src/java/org/apache/cassandra/config/StorageAttachedIndexOptions.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.config;
+
+import com.google.common.base.Objects;
+
+import org.apache.cassandra.exceptions.ConfigurationException;
+
+public class StorageAttachedIndexOptions
+{
+    public static final int DEFAULT_SEGMENT_BUFFER_MB = 1024;
+
+    private static final int MAXIMUM_SEGMENT_BUFFER_MB = 32768;
+
+    public int segment_write_buffer_space_mb = DEFAULT_SEGMENT_BUFFER_MB;
+
+    public void validate()
+    {
+        if ((segment_write_buffer_space_mb < 0) || (segment_write_buffer_space_mb > MAXIMUM_SEGMENT_BUFFER_MB))
+        {
+            throw new ConfigurationException("Invalid value for segment_write_buffer_space_mb. " +
+                                             "Value must be a positive integer less than 32768");
+        }
+    }
+
+    @Override
+    public boolean equals(Object o)
```

Comment:
nit: Do we use `StorageAttachedIndexOptions` as a map key or in a collection? Wondering why the `equals()` and `hashCode()`...

Reply:
I was looking at this the other day and the answer is that we don't so these methods can go.